app/presenters/publishing_api_presenters/link_extractor.rb

```diff
@@ -0,0 +1,77 @@
+module PublishingApiPresenters
+  class LinkExtractor
+    def initialize(item)
+      @item = item
+    end
+
+    # return a hash of all tag types and the values the model contains
+    # Presenters restrict the data sent via publishing api
+    def extract
+      {
+        document_collections: document_collection_ids,
+        lead_organisations: lead_organisation_ids,
+        policy_areas: policy_area_ids,
+        related_policies: related_policy_ids,
+        statistical_data_set_documents: statistical_data_set_ids,
+        supporting_organisations: supporting_organisation_ids,
+        topics: topic_content_ids,
+        world_locations: world_location_ids,
+        worldwide_organisations: worldwide_organisation_ids,
+
+        # FIXME: these tags will be discontinued
+        worldwide_priorities: worldwide_priority_ids,
+      }
+    end
+
+  private
+
+    attr_reader :item
+
+    def document_collection_ids
+      (item.try(:published_document_collections) || []).map(&:content_id)
+    end
+
+    def lead_organisation_ids
+      (item.try(:lead_organisations) || []).map(&:content_id)
+    end
+
+    def policy_area_ids
+      (item.try(:topics) || []).map(&:content_id)
+    end
+
+    def related_policy_ids
+      item.try(:policy_content_ids) || []
+    end
+
+    def statistical_data_set_ids
+      (item.try(:statistical_data_sets) || []).map(&:content_id)
+    end
+
+    def supporting_organisation_ids
+      (item.try(:supporting_organisations) || []).map(&:content_id)
+    end
+
+    def topic_content_ids
+      secondaries = item.try(:secondary_specialist_sector_tags) || []
+      primary = item.try(:primary_specialist_sector_tag)
+      (secondaries.unshift(primary))
+        .compact
+        .map do |tag|
+          Whitehall.content_store.content_item!("/topic/#{tag}").content_id
```

Comment:
Too bad we have to do this, this will be really slow. I'm going to talk to publishing-api people about making an endpoint for `content_id` lookups, which is something we want anyway (https://trello.com/c/RD1ZKLQy).